logprep/util/auto_rule_corpus_tester.py

```diff
@@ -0,0 +1,483 @@
+            self.pipeline._process_pipeline()
+            output = self._retrieve_pipeline_output()
+            current_test_case[1].update({"logprep_output": output})
+
+    def _compare_with_expected_outputs(self):
+        """Compare the generated logprep output with the current test case"""
+        self.console.print("[b]# Test Cases Summary:")
+        for test_case_id in self.test_cases:
+            self._compare_and_collect_report_print_statements(test_case_id)
+            self._print_short_test_result(test_case_id)
+            if len(self.test_cases.get(test_case_id, {}).get("report_print_statements", [])) > 0:
+                self.at_least_one_failed = True
+
+    def _compare_and_collect_report_print_statements(self, test_case_id):
+        test_case_data = self.test_cases.get(test_case_id, {})
+        _, _, logprep_errors = test_case_data.get("logprep_output")
+        if logprep_errors:
+            self.test_cases[test_case_id]["report_print_statements"].extend(
+                ["[red]Following errors happened:", logprep_errors]
+            )
+        if test_case_data.get("test_data_path", {}).get("out"):
+            self._compare_logprep_outputs(test_case_id)
+        if test_case_data.get("test_data_path", {}).get("out_extra"):
+            self._compare_extra_data_output(test_case_id)
+
+    def _print_detailed_reports(self):
+        """If test case reports exist print out each report"""
+        has_failed_reports = any(
+            case[1].get("report_print_statements", False) for case in self.test_cases.items()
+        )
+        if not has_failed_reports:
+            return
+        self.console.print()
+        self.console.print("[b]# Test Cases Detailed Reports:")
+        for test_case_id, test_case_data in self.test_cases.items():
+            if test_case_data.get("report_print_statements"):
+                self._print_long_test_result(test_case_id, test_case_data)
+                self.console.print()
+
+    def _print_test_statistics(self):
+        """Print minimal statistics of the test run"""
+        self.console.print("[b]# Test Overview")
+        total_cases = len(self.test_cases)
+        failed_cases = len(
+            [case for case in self.test_cases.items() if case[1].get("report_print_statements")]
+        )
+        self.console.print(f"Failed tests: {failed_cases}")
+        self.console.print(f"Total test cases: {total_cases}")
+        if total_cases:
+            success_rate = (total_cases - failed_cases) / total_cases * 100
+            self.console.print(f"Success rate: {success_rate:.2f}%")
+
+    def _strip_input_file_type(self, filename):
+        """Remove the input file suffix to identify the case name"""
+        filename = filename.replace("_in", "")
+        filename = filename.replace("_out_extra", "")
+        filename = filename.replace("_out", "")
+        filename = filename.replace(".json", "*")
+        return filename
+
+    def _create_patched_pipeline(self):
```

Comment:
I think this is not ideal. We have to touch it everytime we implement new things features in the pipeline or we refactor things. I think it is better to run logprep as it is for these tests.

Reply:
There is sadly no way around patching a pipeline. The idea of this tester is that I can pass in an original logprep configuration, with for example kafka connectors. To allow for a proper test run this connectors have to be replaced with json connectors. Furthermore the test run here is done with only one process and no metrics, such that these changes have to be patched into the original configuration. I did revise the code though and adjusted the DryRunner such that the DryRun and this RuleCorpusTester both use the same patch pipeline functionality now.